### PR TITLE
Add tests for actions.0.dart API example.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR contributes to https://github.com/flutter/flutter/issues/130459

### Description
- Updates `examples/api/lib/widgets/actions/actions.0.dart` to meet the latest API examples structure
- Adds tests for `examples/api/lib/widgets/actions/actions.0.dart`

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: examples/api/test/widgets/actions/actions.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        matching: find.byWidgetPredicate(
+          (Widget widget) => widget is TextButton,
+        ),
+      ),
+    );
+
+    return button.style?.foregroundColor?.resolve(<WidgetState>{});
+  }
+
+  testWidgets('increments and decrements value', (WidgetTester tester) async {
```
Comment (bleroux) [inline examples/api/test/widgets/actions/actions.0_test.dart:24]: Consider capitalizing the first letter of the test description.

**Review by bleroux — APPROVED** [review]
LGTM! Good call on adding a proper disposal of ValueNotifiers 👍 .
Just one small nit.

**Review by TahaTesser — APPROVED** [review]
LGTM

**Comment by ksokolovskyi** [comment]
@bleroux @TahaTesser thanks for the review and merge!
I fixed the name of the test in my other open PR: https://github.com/flutter/flutter/pull/148676

**Pull request merged**